app/src/main/AndroidManifest.xml

```diff
@@ -68,6 +67,25 @@
             </intent-filter>
         </activity>
         <activity android:name=".recording.RecordingActivity" />
+        <meta-data
+            android:name="com.facebook.sdk.ApplicationId"
+            tools:replace="android:value"
+            android:value="@string/facebook_application_id"
+            />
+        <activity android:name="com.facebook.FacebookActivity"
+            android:configChanges=
+                "keyboard|keyboardHidden|screenLayout|screenSize|orientation"
+            android:label="@string/app_name" />
+        <activity
+            android:name="com.facebook.CustomTabActivity"
+            android:exported="true">
+            <intent-filter>
+                <action android:name="android.intent.action.VIEW" />
+                <category android:name="android.intent.category.DEFAULT" />
+                <category android:name="android.intent.category.BROWSABLE" />
+                <data android:scheme="@string/fb_login_protocol_scheme" />
+            </intent-filter>
+        </activity>
     </application>
-
+    <uses-permission android:name="android.permission.INTERNET" />
```

Comment:
I think (or at least Cirrus does) that you should move the <uses-permission> above the <application>

Reply:
The [documentation] (https://developers.facebook.com/docs/facebook-login/android/) at step 4 specifically states to put this after application that's why I moved it. But further search on internet suggests to place it before as well even though there's no explanation whatsoever. I'll try it like that.